rust/main/hyperlane-base/src/types/gcs_storage.rs

```diff
@@ -109,6 +111,24 @@ impl GcsStorageClient {
     fn get_checkpoint_key(index: u32) -> String {
         format!("checkpoint_{index}_with_id.json")
     }
+
+    fn object_path(&self, object_name: &str) -> String {
+        if let Some(folder) = &self.folder {
+            format!("{}/{}", folder.trim_end_matches('/'), object_name)
```

Comment:
makes sense to `folder.trim_end_matches('/')`, but instead of doing this each time here should it live in `GcsStorageClientBuilder::build` instead? Also worth logging there if anything _is_ trimmed, to avoid any silent unexpected behavior